### PR TITLE
Spelling corrections in markdown files

### PR DESCRIPTION
**Pull request by djyasin**
Spelling corrections in markdown files.

Co-Authored By: Lila Yasin <89486372+djyasin@users.noreply.github.com>

##### SUMMARY

Adding a few code spelling and spelling corrections. Cherry-picked from a community PR #11439

##### ISSUE TYPE

 - New or Enhanced Feature

##### COMPONENT NAME

 - Docs
 - Other

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,6 +54,23 @@ I18N_FLAG_FILE = .i18n_built
 	VERSION PYTHON_VERSION docker-compose-sources \
 	.git/hooks/pre-commit
 
+shellcheck:
```
Comment (fosterseth) [inline Makefile]: @shanemcd  does this makefile diff look good to you? if so we'll merge this in soon

File: tools/docker-compose/entrypoint.sh [hunk 2]
```diff
@@ -32,4 +32,4 @@ else
     export SDB_NOTIFY_HOST=$(ip route | head -n1 | awk '{print $3}')
 fi
 
-exec $@
+exec "$@"
```
Comment (shanemcd) [inline tools/docker-compose/entrypoint.sh]: This change looks suspicious to me and would expect it to cause problems for commands with more than one argument.

**Comment by shanemcd** [comment]
I called this out on the original PR but, right now this seems to rely on someone manually running this command, which doesn't feel particularly useful. It's very trivial to add new CI checks and I think we should do that before merging this PR: https://github.com/ansible/awx/blob/a0b8215c06918260c4f3d0c481fb1adac6fa2613/.github/workflows/ci.yml#L17 The only additional thing you might need to do is ensure the correct package is installed, but only in the _devel_ image. There are other examples in the Dockerfile of us doing this.

**Review by shanemcd — CHANGES_REQUESTED** [review]
Added some comments. The main thing that worries me right now is the change to the entry point. But I also don't see this being very useful without a CI check.

**Review by AlexSCorey — APPROVED** [review]
UI portion looks good to me.

File: Makefile (hunk 1 above)
Comment (gundalow) [reply Makefile]: How come we've got more than detect & fix?

File: Makefile [hunk 3]
```diff
@@ -54,6 +54,23 @@ I18N_FLAG_FILE = .i18n_built
 	VERSION PYTHON_VERSION docker-compose-sources \
 	.git/hooks/pre-commit
```
Comment (gundalow) [inline Makefile]: We will add these into CI at a later point?

**Review by gundalow — COMMENTED** [review]
Do you plan on adding shellcheck & codespell into CI?

**Comment by djyasin** [comment]
This PR had a bit of a mix-up. When we had initially discussed pulling these changes in, we decided not to pull in the Makefile changes and not to add to CI. That decision was made because we realized we would A) have to go back and retroactively fix all the existing spelling issues (I worked on that for a bit and realized it was very big endeavor to sift through) and B) it could cause PRs to fail since sometimes the spellchecks aren't 100% correct.

When I cherry-picked this commit initially, I didn't realize I had grabbed those changes as well.

I have now pulled out the makefile changes and am just keeping the spelling suggestions.

Let me know if you feel it should be done a different way- I've had quite a bit of conversation around this one and there are different opinions. I can definitely pivot!

**Comment by gundalow** [comment]
> This PR had a bit of a mix-up. When we had initially discussed pulling these changes in, we decided not to pull in the Makefile changes and not to add to CI. That decision was made because we realized we would A) have to go back and retroactively fix all the existing spelling issues (I worked on that for a bit and realized it was very big endeavor to sift through) and B) it could cause PRs to fail since sometimes the spellchecks aren't 100% correct.
> 
> When I cherry-picked this commit initially, I didn't realize I had grabbed those changes as well.
> 
> I have now pulled out the makefile changes and am just keeping the spelling suggestions.
> 
> Let me know if you feel it should be done a different way- I've had quite a bit of conversation around this one and there are different opinions. I can definitely pivot!

Thanks for the explanation. 
This sounds good, though we may just need to accept things might regress without CI (though this PR still makes things better than they were).

**Comment by djyasin** [comment]
> > This PR had a bit of a mix-up. When we had initially discussed pulling these changes in, we decided not to pull in the Makefile changes and not to add to CI. That decision was made because we realized we would A) have to go back and retroactively fix all the existing spelling issues (I worked on that for a bit and realized it was very big endeavor to sift through) and B) it could cause PRs to fail since sometimes the spellchecks aren't 100% correct.
> > When I cherry-picked this commit initially, I didn't realize I had grabbed those changes as well.
> > I have now pulled out the makefile changes and am just keeping the spelling suggestions.
> > Let me know if you feel it should be done a different way- I've had quite a bit of conversation around this one and there are different opinions. I can definitely pivot!
> 
> Thanks for the explanation. This sounds good, though we may just need to accept things might regress without CI (though this PR still makes things better than they were).

Thank you!! I will keep this in my notes in case we do decide to take the time to fix all of the existing spelling issues and add the CI check.

**Comment by djyasin** [comment]
@shanemcd does this look okay now?

**Comment by shanemcd** [comment]
I think it's good to fix these typos, but I also think it would be worthwhile and relatively trivial to add a CI check that runs the utility mentioned in https://github.com/ansible/awx/pull/11439. 

If we're not going to do that here, please open an issue for running spell checking as part of the PR checks. It should probably get added to https://github.com/ansible/awx/blob/devel/.github/workflows/ci.yml. 

You can see there is a pretty well established pattern there. As long as the dependency is installed in the devel image, it will just be a matter of adding a new item to the test [matrix](https://github.com/ansible/awx/blob/daf39dc77ecb75a97f7509ffa58ab76b18f4e8a9/.github/workflows/ci.yml#L19).

**Comment by djyasin** [comment]
> It should probably get added to https://github.com/ansible/awx/blob/devel/.github/workflows/ci.yml.

Thank you so much for your feedback! I went ahead and made a new issue here: https://github.com/ansible/awx/issues/14039

**Review by shanemcd — APPROVED** [review]

**Pull request merged**